### DISCUSSION TITLE
Skip ignition if flatcar.first_boot=1 is set

### DISCUSSION THREAD

**Discussion by cullenmcdermott** [flatcar/Flatcar#1001] [Q&A] [answered]
I'm just getting started with Flatcar and booting with iPXE. I currently have the following pxe config

```
kernel ${base-url}/flatcar_production_image.vmlinuz initrd=flatcar_production_pxe_image.cpio.gz flatcar.first_boot=1 ignition.config.url=http://192.168.0.234:8080/pxe-config.ign root=LABEL=ROOT
initrd ${base-url}/flatcar_production_pxe_image.cpio.gz
boot
```
Is there a way to get ignition to no-op if this is not actually the first boot of the VM? I would like to avoid having to flip this flag if possible as there's not an easy way to do it with Terraform. I am considering using Matchbox for this but that would still require two terraform runs.

My setup is with proxmox VMs and I have the boot order set to look at the disk first and then try network. However proxmox hangs at "Booting from hard disk" so I'm hoping there's a way to work around this purely with Flatcar/ignition.

Ultimately I would love for it to always pxe boot and only run ignition the first time it boots without having to flip that flag.

**Comment by cullenmcdermott** [comment, marked as answer]
I think I may have figured out how to get what I want. I set `wipeFilesystem` and `wipeTable` to false so ignition still runs but it doesn't wipe my data so this works! If there was a way to have ignition just no-op that would be ideal but this will probably work for me.

**Comment by tormath1** [comment]
Hi @cullenmcdermott,

Is your plan is to install Flatcar on the disk once PXE booted? Using `flatcar-install` for example? Ignition does run once and only once at the first boot, so as long as there is a `flatcar.first_boot=1` it will run.

**Reply by cullenmcdermott** [reply]
Hi! Yes that was my intention but proxmox not wanting to boot from the disk for some reason seems to prevent that. The setup I have now seems to be working to allow my nodes to reboot and persist their data so I think I’m good. Thank you for the response!

**Reply by tormath1** [reply]
>  proxmox not wanting to boot from the disk for some reason

Do you have any logs to share ?

**Reply by pothos** [reply]
Maybe PXE boot needs to be disabled again at the right time?
